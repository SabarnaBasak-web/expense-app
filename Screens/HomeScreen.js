import React from 'react';
import {StyleSheet, Text, View, SafeAreaView} from 'react-native';
import CardComponent from '../Components/CardComponent';
import {Colors} from '../Colors/Colors';
import Icon from 'react-native-vector-icons/FontAwesome5';

const HomeScreen = () => {
  return (
    <SafeAreaView>
      <View style={styles.container}>
        <CardComponent styled={styles.cardStyle}>
          <View style={styles.icon}>
            <Icon name="wallet" size={35} color={Colors.secondary} />
          </View>
          <View style={styles.header}>
            <Text style={styles.headerText}>Wallet</Text>
          </View>
          <View style={styles.content}>
            <Text style={styles.contentText}>
              {' '}
              <Icon name="rupee-sign" size={14} /> 1000
            </Text>
            <Text style={styles.contentDate}>
              {' '}
              <Icon name="calendar" size={12} /> 12/12/2021
            </Text>
          </View>
        </CardComponent>
        <CardComponent styled={styles.cardStyle}>
          <View style={styles.icon}>
            <Icon name="money-bill-wave" size={35} color={Colors.red400} />
          </View>
          <View style={styles.header}>
            <Text style={styles.headerText}>Expenses</Text>
          </View>
          <View style={styles.content}>
            <Text style={styles.contentText}>
              {' '}
              <Icon name="rupee-sign" size={14} /> 4000
            </Text>
            <Text style={styles.contentDate}>
              {' '}
              <Icon name="calendar" size={12} /> 12/12/2021
            </Text>
          </View>
        </CardComponent>
        <CardComponent styled={styles.cardStyle}>
          <View style={styles.icon}>
            <Icon name="piggy-bank" size={35} color={Colors.teal} />
          </View>
          <View style={styles.header}>
            <Text style={styles.headerText}>Savings</Text>
          </View>
          <View style={styles.content}>
            <Text style={styles.contentText}>
              {' '}
              <Icon name="rupee-sign" size={14} /> 100
            </Text>
            <Text style={styles.contentDate}>
              {' '}
              <Icon name="calendar" size={12} /> 12/12/2021
            </Text>
          </View>
        </CardComponent>
      </View>
    </SafeAreaView>
  );
};

export default HomeScreen;

const styles = StyleSheet.create({
  container: {
    padding: 10,
    marginTop: 5,
  },
  contentText: {
    fontFamily: 'GoogleSans-Regular',
    fontSize: 24,
    fontWeight: 'bold',
  },
  headerText: {
    fontFamily: 'GoogleSans-Bold',
    fontSize: 25,
    color: Colors.black,
  },
  cardStyle: {
    padding: 10,
    display: 'flex',
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
    marginVerticalTop: 5,
    marginBottom: 5,
  },
  content: {
    marginTop: 4,
    padding: 3,
    display: 'flex',
    flex: 1,
    justifyContent: 'flex-end',
    alignItems: 'center',
    alignContent: 'center',
  },
  icon: {
    display: 'flex',
    flex: 0.5,
  },
  header: {
    display: 'flex',
    flex: 1,
    justifyContent: 'flex-start',
  },
  contentDate: {
    fontFamily: 'GoogleSans-Regular',
    fontSize: 12,
    fontWeight: '400',
  },
});
