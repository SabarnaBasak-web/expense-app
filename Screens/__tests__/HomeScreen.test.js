import HomeScreen from '../HomeScreen';
import renderer from 'react-test-renderer';
import React from 'react';
import 'react-native';
import {fireEvent, render} from '@testing-library/react-native';

// test('renders correctly', () => {
//   const tree = renderer.create(<HomeScreen />).toJSON();
//   expect(tree).toMatchSnapshot();
// });

const Button = ({onClick, children}) => {
  <Button onPress={onClick}>Press</Button>;
};

test('button should render', () => {
  const buttonName = 'Press';
  const {getByTestId} = render(<HomeScreen />);
  const foundButton = getByTestId(buttonName);
  expect(foundButton).toBeTruthy();
});

test('button click should show an alert', () => {
  const handleClick = jest.fn();
  render(<Button onPress={handleClick}>Press</Button>);
  fireEvent.click()
});
