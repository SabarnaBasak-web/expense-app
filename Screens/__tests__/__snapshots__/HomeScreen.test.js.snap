// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
>
  <View>
    <Text
      style={
        Object {
          "fontFamily": "GoogleSans-Regular",
        }
      }
    >
       Home Screen Page
    </Text>
    <View
      accessibilityRole="button"
      accessible={true}
      collapsable={false}
      focusable={true}
      nativeID="animatedComponent"
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Press
        </Text>
      </View>
    </View>
    <Text
      style={
        Object {
          "fontFamily": "GoogleSans-Regular",
        }
      }
    >
      Another Text
    </Text>
  </View>
</RCTSafeAreaView>
`;
